src/context.js

```diff
@@ -0,0 +1,21 @@
+import React, { useContext, useState } from "react";
+import data from "./data";
+
+const AppContext = React.createContext()
+const AppProvider = ({ children }) => {
+    const [list, setList] = useState(data)
+    return <AppContext.Provider value={{
+        list,
+        setList
+    }}>
+        {children}
+    </AppContext.Provider>
+}
```

Comment:
It would be better to write separate functions in UserProvider that will return you all users (getAllUsers ()) and one user, with help to id (getUserById (id)) and pass those 2 functions to the provider as value